configscanning/githubrepo.py

```diff
@@ -0,0 +1,302 @@
+"""Tools for use with a specific GitHub repo"""
+
+import logging
+import os
+import shutil
+from pathlib import Path
+from typing import Optional
+from urllib.parse import ParseResult, urlparse
+
+import pygit2
+from filelock import FileLock
+from github import Auth, Github, GithubIntegration
+from github.Repository import Repository
+
+logger = logging.getLogger(__name__)
+
+
+# pygit2 uses the C wrapper and pylint falsely things many things are not defined because of this.
+# pylint: disable=no-member
+class AI4DTERepo:
+    """This represents a cloned repo"""
+
+    location: Path
```

Comment:
do these need to be class attributes? I don't see any class or static methods, so i think they can just be instance attributes (i.e. they don't need to be declared here). If your OOP knowledge isn't strong we can have a chat about this. Or, if there is a reason for this I've not seen then please ignore comment.